packages/realm-react/src/useQuery.tsx

```diff
@@ -29,25 +39,33 @@ import { symbols } from "@realm/common";
  */
 export function createUseQuery(useRealm: () => Realm) {
   return function useQuery<T>(
-    type: string | ({ new (...args: any): T } & Realm.ObjectClass),
-  ): Realm.Results<T & Realm.Object> {
+    ...[type, collectionCallback, dependencies]: UseQueryArgs<T>
```

Comment:
```suggestion
    type: RealmObjectType<T>, 
    query: QueryCallback<T> = (collection) => collection, 
    deps: DependencyList = [queryCallback]
    ): RealmResults<T> {
```
Type the arguments directly.  This will make the typings much cleaner in the code implementing `useQuery`, while also showing `query` and `deps` are optional.

I've also renamed the arguments to what I feel is more consistent with the naming conventions of basic hooks from React.

Reply:
Should the default for `deps` not just be an empty array? Where is `queryCallback` coming from?

Reply:
Typo, `queryCallback` is `query`.
It should be an empty array, but perhaps it would be good to spread `query` into `deps` in the `useMemo`

Reply:
So one goal I had here was to make it so that the end user did not have to define their `query` function with `useCallback`

The `query` param will change on each render since it will be defined inline by the user, but it is memoized internally within the `useQuery` hook based on the dependency array the user passes. Therefore in a design like that we should not rely on `query` to be a stable object

Reply:
Good point 👍🏼